helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/MaxCapacityUsageInstanceConstraint.java

```diff
@@ -37,7 +37,7 @@ protected double getAssignmentScore(AssignableNode node, AssignableReplica repli
       ClusterContext clusterContext) {
     float estimatedMaxUtilization = clusterContext.getEstimatedMaxUtilization();
     float projectedHighestUtilization =
-        node.getGeneralProjectedHighestUtilization(replica.getCapacity());
+        node.getGeneralProjectedHighestUtilization(replica.getCapacity(), clusterContext.getPreferredScoringKey());
```

Comment:
is there always a set preferred scoring key defined? what is the default value?

Reply:
nope, there isn't always a preferred scoring key defined. getPreferredScoringKey() would return null by default.